lib/Dialect/TTIR/Analysis/ShardSolver.cpp

```diff
@@ -37,10 +45,38 @@ ShardSolver::ShardSolver(
       if (operandOp && shardedOps.count(operandOp) > 0) {
         operandOpEdges[op].emplace_back(Edge(operandOp, op, operandIndex));
         userOpEdges[operandOp].emplace_back(Edge(operandOp, op, operandIndex));
+
+        // Mannually insert reshard edges before ShardSolver start
+        // resolving ShardChainConfig
+        //
+        if (!isa<NameLoc>(op->getLoc())) {
+          continue;
+        }
+
+        StringRef opLocName = mlir::cast<NameLoc>(op->getLoc()).getName();
+        auto opInputOverride = inputLayoutOverrides->find(opLocName);
+
+        if (opInputOverride == inputLayoutOverrides->end()) {
+          continue;
+        }
+
+        SmallVector<int64_t> operandIndexes =
+            opInputOverride->getValue().operandIdxes;
+        if (std::find(operandIndexes.begin(), operandIndexes.end(),
+                      operandIndex) != operandIndexes.end()) {
+          insertReshard(Edge(operandOp, op, operandIndex));
+          inputLayoutOverrides->erase(opInputOverride);
```

Comment:
Why erase, we should avoid directly modifying inputs. I would even say this should be passed around as const.